homeassistant/components/scrape/sensor.py

```diff
@@ -84,60 +108,73 @@ async def async_setup_entry(
         )(sensor)
 
         name: str = sensor_config[CONF_NAME]
-        select: str = sensor_config[CONF_SELECT]
-        attr: str | None = sensor_config.get(CONF_ATTRIBUTE)
-        index: int = int(sensor_config[CONF_INDEX])
         value_string: str | None = sensor_config.get(CONF_VALUE_TEMPLATE)
-        unique_id: str = sensor_config[CONF_UNIQUE_ID]
 
         value_template: Template | None = (
             Template(value_string, hass) if value_string is not None else None
         )
+
+        trigger_entity_config = {
+            CONF_NAME: name,
+            CONF_DEVICE_CLASS: sensor_config.get(CONF_DEVICE_CLASS),
+            CONF_UNIQUE_ID: sensor_config[CONF_UNIQUE_ID],
+        }
+
         entities.append(
             ScrapeSensor(
                 hass,
                 coordinator,
-                sensor_config,
-                name,
-                unique_id,
-                select,
-                attr,
-                index,
+                trigger_entity_config,
+                sensor_config.get(CONF_UNIT_OF_MEASUREMENT),
+                sensor_config.get(CONF_STATE_CLASS),
+                sensor_config[CONF_SELECT],
+                sensor_config.get(CONF_ATTRIBUTE),
+                sensor_config[CONF_INDEX],
                 value_template,
+                False,
             )
         )
 
     async_add_entities(entities)
 
 
-class ScrapeSensor(CoordinatorEntity[ScrapeCoordinator], TemplateSensor):
+class ScrapeSensor(
+    CoordinatorEntity[ScrapeCoordinator], ManualTriggerEntity, SensorEntity
+):
     """Representation of a web scrape sensor."""
 
     def __init__(
         self,
         hass: HomeAssistant,
         coordinator: ScrapeCoordinator,
-        config: ConfigType,
-        name: str,
-        unique_id: str | None,
+        trigger_entity_config: ConfigType,
+        unit_of_measurement: str | None,
+        state_class: str | None,
         select: str,
         attr: str | None,
         index: int,
         value_template: Template | None,
+        yaml: bool,
     ) -> None:
         """Initialize a web scrape sensor."""
         CoordinatorEntity.__init__(self, coordinator)
-        TemplateSensor.__init__(
-            self,
-            hass,
-            config=config,
-            fallback_name=name,
-            unique_id=unique_id,
-        )
+        ManualTriggerEntity.__init__(self, hass, trigger_entity_config)
+        self._attr_name = trigger_entity_config[CONF_NAME].template if yaml else None
```

Comment:
Setting the name to None should be in a separate PR together with adding a device, I think?